pahfit/model.py

```diff
@@ -166,11 +170,111 @@ def guess(self, spec: Spectrum1D, redshift=None):
+                # wavelength too long for stellar BB. Evaluate BB at
+                # 5 micron, and spectrum data at minimum wavelength.
+                wsafe = 5
+                amp_guess = sp(w) / bb(wsafe)
+
+            return amp_guess
+
+        loop_over_non_fixed("starlight", "tau", starlight_guess)
+
+        # count number of blackbodies in the model
+        nbb = len(self.features[self.features["kind"] == "dust_continuum"])
+
+        def dust_continuum_guess(row):
+            temp = row["temperature"][0]
+            fmax_lam = 2898.0 / temp
+            bb = BlackBody1D(1, temp)
+            if fmax_lam >= wmin and fmax_lam <= wmax:
+                w = fmax_lam
+                flux_ref = sp(w)
+            elif fmax_lam > wmax:
+                w = wmax
+                flux_ref = yz[np.argmax(xz)]
+            else:
+                w = wmin
+                flux_ref = yz[np.argmin(xz)]
+
+            amp_guess = flux_ref / bb(w)
+            return amp_guess / nbb
+
+        loop_over_non_fixed("dust_continuum", "tau", dust_continuum_guess)
+
+        def line_fwhm_guess(row):
+            w = row["wavelength"][0]
+            if not instrument.within_segment(w, inst):
+                return 0
+
+            fwhm = instrument.fwhm(inst, w, as_bounded=True)[0][0]
+            return fwhm
+
+        def amp_guess(row, fwhm):
+            w = row["wavelength"][0]
+            if not instrument.within_segment(w, inst):
+                return 0
+
+            factor = 1.5
+            wmin = w - factor * fwhm
+            wmax = w + factor * fwhm
+            xz_window = np.logical_and(wmin < xz, xz < wmax)
+            xpoints = xz[xz_window]
+            ypoints = yz[xz_window]
+            if np.count_nonzero(xz_window) >= 2:
+                # difference between flux in window and flux around it
+                power_guess = integrate.trapezoid(yz[xz_window], xz[xz_window])
+                # subtract continuum estimate, but make sure we don't go negative
+                continuum = (ypoints[0] + ypoints[-1]) / 2 * (xpoints[-1] - xpoints[0])
+                if continuum < power_guess:
+                    power_guess -= continuum
+            else:
+                power_guess = 0
+
+            return power_guess / fwhm
+
+        # calc line amplitude using instrumental fwhm and integral over data
+        loop_over_non_fixed(
+            "line", "power", lambda row: amp_guess(row, line_fwhm_guess(row))
+        )
+        # set the fwhms in the features table requested
+        if calc_line_fwhm:
+            loop_over_non_fixed("line", "fwhm", line_fwhm_guess, force=True)
+
+        def df_amp_guess(row):
+            w = row["wavelength"][0]
+            fwhm=row["fwhm"][0]
+            if not (instrument.within_segment(w - fwhm, inst) or instrument.within_segment(w + fwhm, inst)):
+                return 0
+
+            drude = Drude1D(amplitude=1, x_0=w, fwhm=fwhm)
+            return sp(w) / drude(w)
```

Comment:
This seems a bit overly complex.  E.g. why not just evaluate at `x_0` (where the Drude will peak)?

Reply:
I see, Drude1D evaluated at x_0 is always equal to amplitude. Will change when we have power-based drudes though.

Reply:
I replaced this with the old method (setting amplitude equal to median flux), since trying to be smart makes the fitter prone to getting stuck in local minima.